data-prepper-plugins/sns-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/dlq/DlqPushHandler.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.dlq;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import org.opensearch.dataprepper.metrics.MetricNames;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.failures.DlqObject;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.plugins.dlq.DlqProvider;
+import org.opensearch.dataprepper.plugins.dlq.DlqWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static java.util.UUID.randomUUID;
+
+
+/**
+ * * An Handler class which helps log failed data to AWS S3 bucket or file based on configuration.
+ */
+
+public class DlqPushHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DlqPushHandler.class);
+
+    private static final String BUCKET = "bucket";
+
+    private static final String ROLE_ARN = "sts_role_arn";
+
+    private static final String REGION = "region";
+
+    private static final String S3_PLUGIN_NAME = "s3";
+
+    private static final String KEY_PATH_PREFIX = "key_path_prefix";
+
+    private String dlqFile;
+
+    private String keyPathPrefix;
+
+    private DlqProvider dlqProvider;
+
+    public DlqPushHandler(final String dlqFile,
+                          final PluginFactory pluginFactory,
+                          final String bucket,
+                          final String stsRoleArn,
+                          final String awsRegion,
+                          final String dlqPathPrefix) {
+        if(dlqFile != null) {
+            this.dlqFile = dlqFile;
+        }else{
+            this.dlqProvider = getDlqProvider(pluginFactory,bucket,stsRoleArn,awsRegion,dlqPathPrefix);
+        }
+    }
+
+    public void perform(final PluginSetting pluginSetting,
+                        final Object failedData) {
+        if(dlqFile != null)
+            writeToFile(failedData);
+        else
+            pushToS3(pluginSetting, failedData);
+    }
+
+    private void writeToFile(Object failedData) {
+        try(BufferedWriter dlqFileWriter = Files.newBufferedWriter(Paths.get(dlqFile),
+                StandardOpenOption.CREATE, StandardOpenOption.APPEND)) {
+            dlqFileWriter.write(failedData.toString());
+        } catch (IOException e) {
+            LOG.error("Exception while writing failed data to DLQ file Exception: ",e);
+        }
+    }
+
+    private void pushToS3(PluginSetting pluginSetting, Object failedData) {
+        LOG.info("s3 dlq - push to s3 method");
```

Comment:
This seems more like debug logs

Reply:
Thanks @graytaylor0 for your feedback. yes, used for debuging and changed to debug.